src/session-managers/SecureSessionManager.ts

```diff
@@ -20,14 +20,25 @@ export class SecureSessionManager {
 
   async logIn(request: FastifyRequest, user: any) {
     const object = await this.serializeUser(user, request)
-    request.session.set(this.key, object)
+    if (typeof request.session.set === 'function') {
```

Comment:
Could we do a different session manager class instead of one that has a bunch of if statements in it? It feels a bit weird to duck type on the session object like this

Reply:
I'm +1 on create a new session manager for `@fastify/session`.

Reply:
How to determine which session manager class should be used?

Reply:
@AaronPorts you can take https://github.com/fastify/fastify-csrf/ as an example.